frontend/pages/admin/IntegrationsPage/cards/ConditionalAccess/ConditionalAccess.tsx

```diff
@@ -40,14 +40,31 @@ const MSETID = "microsoft_entra_tenant_id";
 interface IDeleteConditionalAccessModal {
   toggleDeleteConditionalAccessModal: () => void;
   onDelete: () => void;
-  isUpdating: boolean;
 }
 
 const DeleteConditionalAccessModal = ({
   toggleDeleteConditionalAccessModal,
   onDelete,
-  isUpdating,
 }: IDeleteConditionalAccessModal) => {
+  const { renderFlash } = useContext(NotificationContext);
+  const [isDeleting, setIsDeleting] = useState(false);
+
+  const handleDelete = async () => {
```

Comment:
Why split this logic into the passed-in `onDelete` and this `handleDelete` instead of having it all
in the parent `onDelete` like before? Could you just add the `setFormData({ [MSETID]: "" });` to
that handler as it was and achieve the same end result?

Reply:
> Could you just add the setFormData({ [MSETID]: "" }); to that handler as it was and achieve the same end result?

Actually yes. The clearing of the tenant ID was easily achieved by that.
But what was not working with the original approach was the spinner during the deletion. To make it work I emulated what the other Delete buttons in the app are doing (with its own `useState`).